phylo/src/evolutionary_models/mod.rs

```diff
@@ -61,34 +79,55 @@ impl Display for ProteinModelType {
     }
 }
 
-pub trait EvolutionaryModelParameters<T> {
-    fn new(model_type: &T, params: &[f64]) -> Result<Self>
+impl ProteinModelType {
+    pub fn get_model_type(model_name: &str) -> Self {
+        match model_name.to_uppercase().as_str() {
+            "WAG" => ProteinModelType::WAG,
+            "BLOSUM" => ProteinModelType::BLOSUM,
+            "HIVB" => ProteinModelType::HIVB,
+            _ => {
+                warn!("Unknown DNA model requested, defaulting to WAG.");
+                ProteinModelType::WAG
+            }
+        }
+    }
+}
+
+pub trait EvoModelParams {
+    type ModelType;
+    type Parameter;
+    fn new(model: &Self::ModelType, params: &[f64]) -> Result<Self>
     where
         Self: Sized;
-    fn get_value(&self, param_name: &Parameter) -> f64;
-    fn set_value(&mut self, param_name: &Parameter, value: f64);
+    fn parameter_definition(
+        model_type: &Self::ModelType,
+    ) -> Vec<(&'static str, Vec<Self::Parameter>)>;
+    fn get_value(&self, param_name: &Self::Parameter) -> f64;
```

Comment:
I believe in Rust getters don't have `get_` prefix, so in this case it should be just `value`
https://users.rust-lang.org/t/best-practices-on-setters-and-accessor-methods-in-general/66530/4

Reply:
Since you have parameters and frequences, it absolutely would make sense to name these getters smth like `param` (or `param_value`) and `pi` (or probably `freq`)

Reply:
Done. While I was at it, I removed all "get" from methods — there were a lot. I also renamed pi() to freqs() to avoid confusing it with p() (and there is also already a q()).